controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -239,7 +239,36 @@ bool master_thread::handle_cmdu_1905_autoconfiguration_search(Socket *sd,
 
     LOG(DEBUG) << "Received AP_AUTOCONFIGURATION_SEARCH_MESSAGE";
 
-    auto tlvAlMacAddressType = cmdu_rx.addClass<ieee1905_1::tlvAlMacAddressType>();
+    std::shared_ptr<ieee1905_1::tlvAlMacAddressType> tlvAlMacAddressType     = nullptr;
+    std::shared_ptr<ieee1905_1::tlvSearchedRole> tlvSearchedRole             = nullptr;
+    std::shared_ptr<ieee1905_1::tlvAutoconfigFreqBand> tlvAutoconfigFreqBand = nullptr;
+    std::shared_ptr<wfa_map::tlvSupportedService> tlvSupportedServiceIn      = nullptr;
+    std::shared_ptr<wfa_map::tlvSearchedService> tlvSearchedService          = nullptr;
+    int type = cmdu_rx.getNextTlvType();
```

Comment:
you already assign `type` in each while loop below, so this is redundant, but keep reading the next comment.